### PR TITLE
Show full payload in source debugger

### PR DESCRIPTION
**Pull request by prabrishac**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: services/source-debugger/eventUploader.go [hunk 1]
```diff
@@ -22,25 +22,11 @@ type GatewayEventBatchT struct {
...
-	SentAt            string      `json:"sentAt,omitempty"`
-	Type              string      `json:"type,omitempty"`
-}
-
 //EventUploadBatchT is a structure to hold batch of events
 type EventUploadBatchT struct {
 	WriteKey   string
 	ReceivedAt string
-	Batch      []EventUploadT
+	Batch      []map[string]interface{}
```
Comment (SumanthPuram) [inline services/source-debugger/eventUploader.go:29]: Define a type for EventUpload.
`type EventUploadT map[string]interface{}`

Comment (SumanthPuram) [reply services/source-debugger/eventUploader.go:29]: Makes all further references more readable.

Comment (prabrishac) [reply services/source-debugger/eventUploader.go]: done.

File: services/source-debugger/eventUploader.go [hunk 2]
```diff
@@ -137,17 +126,20 @@ func uploadEvents(eventBuffer []*GatewayEventBatchT) {
 
 			//updating originalTimestamp in the event using the formula
 			//receivedAt - (sentAt - originalTimeStamp)
-			orgTS, err := time.Parse(time.RFC3339, ev.OriginalTimestamp)
+			originalTimestamp := ev["originalTimestamp"].(string)
```
Comment (psrikanth88) [inline services/source-debugger/eventUploader.go:129]: check if event has originalTimestamp property. Only events sent with SDK's have originalTimestamp. And maybe older SDK's do not have it also.
Also handle casting to string error?

File: services/source-debugger/eventUploader.go [hunk 3]
```diff
@@ -137,17 +126,20 @@ func uploadEvents(eventBuffer []*GatewayEventBatchT) {
...
+			sentAt := ev["sentAt"].(string)
```
Comment (psrikanth88) [inline services/source-debugger/eventUploader.go:130]: same err handling as above

Comment (psrikanth88) [reply services/source-debugger/eventUploader.go:130]: also maybe add sentAt in gateway if not present in payload
something like we do for receivedAt https://github.com/rudderlabs/rudder-server/blob/master/gateway/gateway.go#L395 but with additional if present check

File: services/source-debugger/eventUploader.go (hunk 2 above)
Comment (prabrishac) [reply services/source-debugger/eventUploader.go]: done. reusing the same logic from processor.go

File: services/source-debugger/eventUploader.go (hunk 3 above)
Comment (prabrishac) [reply services/source-debugger/eventUploader.go]: done. reusing the same logic from processor.go. As discussed, did not add it in gateway for now.

**Comment by prabrishac** [comment]
Corresponding config BE change: https://github.com/rudderlabs/rudder-config-backend/pull/217

**Pull request merged**